tests/tools/visualization/test_graph.py

```diff
@@ -325,3 +323,36 @@ def test_graph_graphviz_source_pstate(self):
         assert sorted([line.strip() for line in graph.graphviz.source.splitlines()]) == sorted(
             [line.strip() for line in expected.splitlines()]
         )
+
+    @pytest.mark.parametrize(
+        'node_id_type',
+        (
+            'pk',
+            'uuid',
+            'label',
+            ('pk', 'uuid'),
+            ('pk', 'label'),
```

Comment:
Rule of induction :) Show that it holds for `n` and then for `n+1`, it would follow that it works for all `n`. I could have added it for 3 as well, but I thought that if it works for 1 and 2, that should be sufficient. No other reason really

Reply:
Yeah, that's what I thought :D great!